src/components/structures/FavouriteMessagesView/FavouriteMessagesTilesList.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixClient, MatrixEvent } from "matrix-js-sdk/src/matrix";
+import React, { useCallback } from "react";
+
+import { _t } from "../../../languageHandler";
+import Spinner from "../../views/elements/Spinner";
+import FavouriteMessageTile from "./FavouriteMessageTile";
+
+interface IProps {
+    favouriteMessageEvents: MatrixEvent[] | null;
+    favouriteMessagesPanelRef: any;
+    searchQuery: string;
+    cli: MatrixClient;
+}
+
+const FavouriteMessagesTilesList = ({
+    cli,
+    favouriteMessageEvents,
+    favouriteMessagesPanelRef,
+    searchQuery,
+}: IProps) => {
+    const ret: JSX.Element[] = [];
+    let lastRoomId: string;
+    const highlights: string[] = [];
+
+    // once dynamic content in the favourite messages panel loads, make the scrollPanel check
+    // the scroll offsets.
+    const onHeightChanged = useCallback(() => {
+        const scrollPanel = favouriteMessagesPanelRef.current;
+        if (scrollPanel) {
+            scrollPanel.checkScroll();
+        }
+    }, [favouriteMessagesPanelRef]);
+
+    if (!favouriteMessageEvents) {
+        ret.push(<Spinner key="spinner" />);
+    } else {
+        favouriteMessageEvents.forEach((mxEvent) => {
+            const timeline = [] as MatrixEvent[];
+            const roomId = mxEvent.getRoomId();
+            const room = cli?.getRoom(roomId);
+
+            timeline.push(mxEvent);
+            if (searchQuery) {
+                highlights.push(searchQuery);
+            }
+
+            if (roomId !== lastRoomId) {
+                ret.push(
+                    <li key={mxEvent.getId() + "-room"}>
+                        <h2>
+                            {_t("Room")}: {room ? room.name : ""}
+                        </h2>
+                    </li>,
+                );
+                lastRoomId = roomId!;
+            }
+
+            const resultLink = "#/room/" + roomId + "/" + mxEvent.getId();
```

Comment:
this appears to be an assumption of our url routing which isn't guaranteed - can we convert it to a `view_room` dispatch instead?

Reply:
It's copied from here: https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/components/structures/RoomSearchView.tsx#L255

I presume that could do with fixing too?

Reply:
Looking into the code, it seems that `EventTile` does not currently support an `onClick` or similar event, so the only way to allow clicking a tile to jump somewhere else is with its `highlightLink` property as we have done here and in `RoomSearchView`.

Given how long this PR has festered, I'd like to argue that fixing this is outside of the scope of this PR, and something we can do later.